sw/device/tests/spi_host_config_test.c

```diff
@@ -0,0 +1,105 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#include <assert.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_spi_host.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/spi_device_testutils.h"
+#include "sw/device/lib/testing/spi_flash_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ottf_utils.h"
+#include "sw/device/tests/spi_host_flash_test_impl.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__,
+              "This test assumes the target platform is little endian.");
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kDefaultTimeoutMicros = 50000,
+};
+
+uint8_t backdoor_cpha = UINT8_MAX;
+uint8_t backdoor_cpol = UINT8_MAX;
+uint32_t backdoor_data = UINT32_MAX;
+
+static status_t spi_config_test(dif_spi_host_t *spi);
+
+static status_t configure_pinmux(const dif_pinmux_t *pinmux);
+
+bool test_main(void) {
+  dif_spi_host_t spi_host;
+  mmio_region_t base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR);
+
+  dif_pinmux_t pinmux;
+  CHECK_DIF_OK(dif_pinmux_init(base_addr, &pinmux));
+  base_addr = mmio_region_from_addr(TOP_EARLGREY_SPI_HOST1_BASE_ADDR);
+  CHECK_DIF_OK(dif_spi_host_init(base_addr, &spi_host));
+
+  configure_pinmux(&pinmux);
+  for (size_t i = 0; i < 4; ++i) {
+    CHECK_STATUS_OK(spi_config_test(&spi_host));
+  }
+  return true;
+}
+
+static status_t spi_config_test(dif_spi_host_t *spi) {
+  dif_spi_host_config_t config = {
+      .spi_clock = 40000,
```

Comment:
Can you elaborate a bit on this number? 40k seems rather low compared to the values we use in other tests. Is this maybe a typo? I am also curious how this relates to previous comments in the issue regarding limitations on the max speed (16 MHz or 30 MHz). Sorry for asking this, I am no SPI expert :-/

Reply:
I'm intentionally using 40kHz because of a limitation of the test platform, We don't have a spi device on the hyperdebug to test the OT SPI host, so I'm using bit banging instead.To ensure accurate signal capture and avoid aliasing, I've chosen a test frequency of 40kHz, adhering to the Nyquist limit with the Hyperdebug's 100kHz sampling rate.
Does that make sense?

Reply:
Please could you lift this number up into a constant and document it if you get the chance?

Reply:
Thanks @engdoreis for providing more details. This makes perfect sense to me! It may be worth adding this information in a comment in case you decide to add a constant as @jwnrt recommended.